_posts/2018-04-09-synching-sia.md

```diff
@@ -0,0 +1,131 @@
+
+My 73-minute sync only spent 168 seconds downloading blocks from peers.  This number is actually slightly confusing to me since if Sia downloaded 10GB of blocks it exceeded my advertised home internet speed.  A couple of things could have happened here:
+The downloaded blocks could be less than 10GB in size when sent over the Internet, I never measured how much data I actually received from my NIC.
+I could have a mistake in my timing function causing the elapsed download time to be reported wrong.
+My ISP could allow me to burst into higher speed tiers during low network congestion.
+
+Regardless of why my download speed seems off, I was able to sync from block 0 to current height in 73 minutes.
+
+The timeouts I set in `synchronize.go` for my 73-minute sync.
+
+```go
+// relayHeaderTimeout is the timeout for the RelayHeader RPC.
+relayHeaderTimeout = build.Select(build.Var{
+        Standard: 10 * time.Second,
+        Dev:      20 * time.Second,
+        Testing:  3 * time.Second,
+}).(time.Duration)
+
+// sendBlkTimeout is the timeout for the SendBlocks RPC.
+sendBlkTimeout = build.Select(build.Var{
+        Standard: 10 * time.Second,
+        Dev:      30 * time.Second,
+        Testing:  4 * time.Second,
+}).(time.Duration)
+
+// sendBlocksTimeout is the timeout for the SendBlocks RPC.
+sendBlocksTimeout = build.Select(build.Var{
+        Standard: 10 * time.Second,
+        Dev:      40 * time.Second,
+        Testing:  5 * time.Second,
+}).(time.Duration)
+
+```
+
+## Sync to a RAMDisk
+
+Validating and Applying the transactions in Sia’s blockchain is a very I/O intensive process. To reduce the amount of time spent doing this I/O I created a 11GB RAMDisk and synced the blockchain to the RAMDisk.
+
+Command to create an 11 GB RAMDisk and mount it at `Sia/consensus`.
+
+```shell
+root@v132:~# mount -t tmpfs -o size=11000m tmpfs Sia/consensus/
+root@v132:~# df -h
+Filesystem                        Size  Used Avail Use% Mounted on
+/dev/mapper/pve-vm--200--disk--1   32G  2.1G   28G   7% /
+none                              492K     0  492K   0% /dev
+tmpfs                              32G     0   32G   0% /dev/shm
+tmpfs                              32G  8.2M   32G   1% /run
+tmpfs                             5.0M     0  5.0M   0% /run/lock
+tmpfs                              32G     0   32G   0% /sys/fs/cgroup
+tmpfs                              11G     0   11G   0% /root/Sia/consensus
+```
+
+A RAMDisk has very fast I/O but is not persistent across reboots.  Most users are going to want to copy the `consensus.db` file to persistent storage after the sync is done.
+
+## Future optimizations
```

Comment:
These two feel weird paired together. The first one I feel like falls in the scope of things the user could trivially do on their own if they want to. The second is something that the devs probably have to spend months doing.

The first one fits with this article, but I think the reader might wonder why the experiment skipped this seemingly easy optimization. Can we explain that?

The second one, consider cutting because it feels outside the scope of the article.

Reply:
I've clarified the 1st point (it requires network adoption before it would have any effect).  I still think these are related because they are both areas that the sync time could be improved, but they require dev time, testing, and network adoption before it would make any difference.

I wanted to include `a consensus overhaul` section because the devs have talked about it and have several ideas on how to improve it which are in the issue I linked.